native/common/jp_bytetype.cpp

```diff
@@ -237,19 +229,29 @@ JPPyObject JPByteType::getArrayItem(JPJavaFrame& frame, jarray a, jsize ndx)
 	frame.GetByteArrayRegion(array, ndx, 1, &val);
 	jvalue v;
 	field(v) = val;
-	return convertToPythonObject(v);
+	return convertToPythonObject(frame, v);
 }
 
 void JPByteType::setArrayItem(JPJavaFrame& frame, jarray a, jsize ndx, PyObject* obj)
 {
-	array_t array = (array_t) a;
-	type_t val = field(convertToJava(obj));
-	frame.SetByteArrayRegion(array, ndx, 1, &val);
+	JPMatch match;
+	if (getJavaConversion(&frame, match, obj) < JPMatch::_implicit)
+		JP_RAISE(PyExc_TypeError, "Unable to convert to Java byte");
+	type_t val = field(match.conversion->convert(&frame, this, obj));
+	frame.SetByteArrayRegion((array_t) a, ndx, 1, &val);
 }
 
+string JPByteType::asString(jvalue v)
+{
+	std::stringstream out;
+	out << (int) v.b;
```

Comment:
Why do you need to explicitly cast this to int? I would assume the b value in the union is already of some integral type capable of representing bytes.

Reply:
The cast to int here was to so it printed a number rather than the null char or one char.

As JNI defines it the union member is a char which unfortunately does the wrong behavior when going to a stream.  

This particular method was routed arround later and got left as dead code.  It is removed in the instrumentation PR.  
Instead of implementing our own i managed to route to Pythons internal method which is a much cleaner solution.

Sorry about the dead code.  When i had to apply the patches out of order (speed/array then jni/classhints) it was a rather painfull remerge and left dead code from both branches.